src/a11y-dialog.js

```diff
@@ -1,283 +1,237 @@
- * Destroy the current instance (after making sure the dialog has been hidden)
- * and remove all associated listeners from dialog openers and closers
- *
- * @return {A11yDialog}
- */
-A11yDialog.prototype.destroy = function () {
-  // Hide the dialog to avoid destroying an open instance
-  this.hide()
-
-  // Remove the click event listener from all dialog openers
-  this._openers.forEach(
-    function (opener) {
-      opener.removeEventListener('click', this._show)
-    }.bind(this)
-  )
-
-  // Remove the click event listener from all dialog closers
-  this._closers.forEach(
-    function (closer) {
-      closer.removeEventListener('click', this._hide)
-    }.bind(this)
-  )
-
-  // Execute all callbacks registered for the `destroy` event
-  this._fire('destroy')
-
-  // Empty the listeners map
-  this._listeners = {}
-
-  return this
-}
+    // Keep a reference to the currently focused element to be able to restore
+    // it later
+    this.previouslyFocused = document.activeElement
 
-/**
- * Register a new callback for the given event type
- *
- * @param {string} type
- * @param {Function} handler
- * @return {A11yDialog}
- */
-A11yDialog.prototype.on = function (type, handler) {
-  if (typeof this._listeners[type] === 'undefined') {
-    this._listeners[type] = []
-  }
+    this.$el.removeAttribute('aria-hidden')
+    this.shown = true
 
-  this._listeners[type].push(handler)
+    // Set the focus to the dialog element
+    moveFocusToDialog(this.$el)
 
-  return this
-}
+    // Bind a focus event listener to the body element to make sure the focus
+    // stays trapped inside the dialog while open, and start listening for some
+    // specific key presses (TAB and ESC)
+    document.body.addEventListener('focus', this.maintainFocus, true)
+    document.addEventListener('keydown', this.bindKeypress)
 
-/**
- * Unregister an existing callback for the given event type
- *
- * @param {string} type
- * @param {Function} handler
- * @return {A11yDialog}
- */
-A11yDialog.prototype.off = function (type, handler) {
-  var index = (this._listeners[type] || []).indexOf(handler)
+    // Execute all callbacks registered for the `show` event
+    this.fire('show', event)
 
-  if (index > -1) {
-    this._listeners[type].splice(index, 1)
+    return this
   }
+  /**
+   * Hide the dialog element, restore the focus to the previously
+   * active element, stop listening for some specific key presses
+   * and fire all registered callbacks for `hide` event.
+   * @param {Event} hide
+   */
+  hide = event => {
+    // If the dialog is already closed, abort
+    if (!this.shown) {
+      return this
+    }
+    this.shown = false
+    this.$el.setAttribute('aria-hidden', 'true')
+
+    // If there was a focused element before the dialog was opened (and it has a
+    // `focus` method), restore the focus back to it
+    // See: https://github.com/KittyGiraudel/a11y-dialog/issues/108
+    if (this.previouslyFocused && this.previouslyFocused.focus) {
```

Comment:
I don’t know if it’s worth adding a Babel plugin for that (unless what you‘ve set up already supports it), but maybe we could do `this.previouslyFocused?.focus()`?